code/utils_model.py

```diff
@@ -1,479 +1,509 @@
             transforms.ToTensor(),
-            transforms.Normalize(
-                PATH_MEAN, PATH_STD
-            )  # mean and standard deviations for lung adenocarcinoma resection slides
+            transforms.Normalize(mean=config.path_mean, std=config.path_std)
         ]),
-        'val':
-        transforms.Compose([
-            transforms.CenterCrop(config.patch_size),
+        "val":
+        transforms.Compose(transforms=[
             transforms.ToTensor(),
-            transforms.Normalize(PATH_MEAN, PATH_STD)
-        ]),
-        'unnormalize':
-        transforms.Compose(
-            [transforms.Normalize(*reverse_norm(PATH_MEAN, PATH_STD))]),
+            transforms.Normalize(mean=config.path_mean, std=config.path_std)
+        ])
     }
 
-    return data_transforms
-
 
-# printing the model
-def print_params(train_folder, num_epochs, num_layers, learning_rate,
-                 batch_size, weight_decay, learning_rate_decay,
-                 resume_checkpoint, resume_checkpoint_path, save_interval,
-                 checkpoints_folder, pretrain, log_csv):
-
-    print("train_folder:", train_folder)
-    print("num_epochs:", num_epochs)
-    print("num_layers:", num_layers)
-    print("learning_rate:", learning_rate)
-    print("batch_size:", batch_size)
-    print("weight_decay:", weight_decay)
-    print("resume_checkpoint:", resume_checkpoint)
-    print("resume_checkpoint_path (only if resume_checkpoint is true):",
-          resume_checkpoint_path)
-    print("save_interval:", save_interval)
-    print("output in checkpoints_folder:", checkpoints_folder)
-    print("pretrain:", pretrain)
-    print("log_csv:", log_csv)
-    print()
+def print_params() -> None:
+    """
+    Print the configuration of the model.
+    """
+    print(f"train_folder: {config.args.train_folder}\n"
+          f"num_epochs: {config.args.num_epochs}\n"
+          f"num_layers: {config.args.num_layers}\n"
+          f"learning_rate: {config.args.learning_rate}\n"
+          f"batch_size: {config.args.batch_size}\n"
+          f"weight_decay: {config.args.weight_decay}\n"
+          f"learning_rate_decay: {config.args.learning_rate_decay}\n"
+          f"resume_checkpoint: {config.args.resume_checkpoint}\n"
+          f"resume_checkpoint_path (only if resume_checkpoint is true): "
+          f"{config.resume_checkpoint_path}\n"
+          f"save_interval: {config.args.save_interval}\n"
+          f"output in checkpoints_folder: {config.args.checkpoints_folder}\n"
+          f"pretrain: {config.args.pretrain}\n"
+          f"log_csv: {config.log_csv}\n\n")
 
 
 ###########################################
 #          MAIN TRAIN FUNCTION            #
 ###########################################
 
 
-# helper function for training resnet
-def train_helper(model, dataloaders, device, dataset_sizes, criterion,
-                 optimizer, scheduler, num_epochs, save_interval, writer):
-
+def train_helper(model: torchvision.models.resnet.ResNet,
```

Comment:
i've never used type enforcement in python, but it this is what it is, then i love it

Reply:
In typical Python fashion, the types are never enforced, but an IDE will warn you if things are the wrong type.